doc/scale_up/Ethernet_Scale_Up_AI_Cluster_Architecture.md

```diff
@@ -0,0 +1,644 @@
+
+        *   Add functionality to handle Config DB events for LLR and CBFC configuration updates, to enable or disable UE link negotiation.
+
+    *   **lldpd Enhancement**:
+
+        *   Implement dynamic LLR/CBFC configuration negotiation using custom TLV (Type-Length-Value) extensions.
+
+    *   **lldp-syncd Enhancement**:
+
+        *   Update Application DB with LLR/CBFC enable/disable status based on Config DB settings and negotiation outcomes.
+
+4. **APP DB**
+
+    define new schema for LLR/CBFC/PFC
+
+
+    ```plain
+    ; Defines information for port configuration
+    key                     = PORT_TABLE:port_name           ; configuration of the port
+    ; field                 = value
+    ...
+    FEC                     = STRING             ; FEC configuration(RS-272/RS-544/autoneg)
+    LLR                     = STRING             ; LLR configuration(on/off)
+    CFBC                    = STRING             ; CFBC configuration(on/off)
+    ```
+
+5. **SWSS** 
+
+     *   Either develop a new OrchAgent or extend PortOrch to program LLR profile and CBFC VC (Virtual Channel) settings into SAI port objects.
+
+      *   Get port LLR/CBFC status/counters from SAI, and update State DB and Counter DB correspondingly
+
+6.  **State DB**
+
+    ```plain
+    ; Defines information for port state
+    key                     = PORT_TABLE:port_name      ; state of the port
+    ; field                 = value
+    ...
+    FEC                     = STRING         ; operational fec mode
+    LLR_TX                  = STRING         ; operational LLR TX status
+    LLR_RX                  = STRING         ; operational LLR RX status
+    LLR_FRAME_ACTION        = STRING         ; operational LLR frame action
+    CFBC                    = STRING         ; operational CFBC status
+    ```
+
+    Here is the table to map the LLR fields and SAI attributes:
+
+    | **Parameter** | **sai\_port\_attr\_t** |
+    | --- | --- |
+    | tx\_off | SAI\_PORT\_LLR\_TX\_STATUS\_OFF |
+    | tx\_init | SAI\_PORT\_LLR\_TX\_STATUS\_INIT |
+    | tx\_adv | SAI\_PORT\_LLR\_TX\_STATUS\_ADVANCE |
+    | tx\_replay | SAI\_PORT\_LLR\_TX\_STATUS\_REPLAY |
+    | tx\_flush | SAI\_PORT\_LLR\_TX\_STATUS\_FLUSH |
+    | ... | ... |
+
+    Here is the table to map the CBFC fields and SAI attributes:
+
+    | **Parameter** | **sai\_port\_attr\_t** |
+    | --- | --- |
+    | credit\_used | SAI\_PORT\_STAT\_CBFC\_SENDER\_CREDITS\_USED |
+    | credit\_watermark | SAI\_PORT\_STAT\_CBFC\_SENDER\_CREDITS\_USED\_WATERMARK |
+    | cc\_msg\_tx | SAI\_PORT\_STAT\_CBFC\_NUM\_CC\_UPDATE\_MESSAGES\_TX |
+    | cf\_msg\_tx | SAI\_PORT\_STAT\_CBFC\_NUM\_CF\_UPDATE\_MESSAGES\_TX |
+    | cc\_msg\_rx | SAI\_PORT\_STAT\_CBFC\_NUM\_CC\_UPDATE\_MESSAGES\_RX |
+    | cf\_msg\_rx | SAI\_PORT\_STAT\_CBFC\_NUM\_CF\_UPDATE\_MESSAGES\_RX |
+
+7.  **SAI**
+
+  *   **LLR Implementation**:
+
+        *   [GitHub Pull Request #1](https://github.com/rck-innovium/SAI/pull/1)
```

Comment:
Please use this link: https://github.com/opencomputeproject/SAI/pull/2225

Reply:
> Please use this link: [opencomputeproject/SAI#2225](https://github.com/opencomputeproject/SAI/pull/2225)

Got it. Thanks!